src/core/service.c

```diff
@@ -3436,9 +3436,18 @@ static void service_notify_cgroup_empty_event(Unit *u) {
         }
 }
 
-static void service_notify_cgroup_oom_event(Unit *u) {
+static void service_notify_cgroup_oom_event(Unit *u, bool managed_oom) {
         Service *s = SERVICE(u);
 
+        if (managed_oom) {
+                log_unit_debug(u, "Process(es) of control group were killed by systemd-oomd.");
+
+                if (s->exit_type == SERVICE_EXIT_MAIN || cgroup_good(s) <= 0)
```

Comment:
Doesn't `|| cgroup_good(s) <= 0` imply that we will set oomd-kill for `ExitType=cgroup` too? nit picking on this detail since in your commit msg you wrote:
> for now we only set the oomd-kill result on services with ExitType=main

Reply:
Got it from core/service.c#L3579, I misunderstood the comment. 
I'll put up a patch fixing this and adding additional xattr. Thanks!